core/engine.go

```diff
@@ -27,13 +27,85 @@ import (
 	"github.com/spf13/pflag"
 )
 
-// EngineCtl is the control structure where engines are registered. All registered engines are referenced by the EngineCtl
-type EngineControl struct {
-	// Engines is the slice of all registered engines
-	Engines []*Engine
+// NewSystem creates a new, empty System.
+func NewSystem() *System {
+	return &System{engines: []*Engine{}}
 }
 
-var EngineCtl EngineControl
+// System is the control structure where engines are registered.
+type System struct {
+	// engines is the slice of all registered engines
+	engines []*Engine
+}
+
+// Diagnostics returns the compound diagnostics for all engines.
+func (system *System) Diagnostics() []DiagnosticResult {
+	result := make([]DiagnosticResult, 0)
+	system.VisitEngines(func(engine *Engine) {
+		if engine.Diagnosable != nil {
```

Comment:
we could also go to a system where you check if an engine can be cast to a Diagnosable? Removing the need to register it in the engine struct?

Reply:
See comment below